hardware_interface/test/fake_components/test_generic_system.cpp

```diff
@@ -323,8 +358,78 @@ class TestGenericSystem : public ::testing::Test
   std::string hardware_system_2dof_standard_interfaces_with_offset_;
   std::string hardware_system_2dof_standard_interfaces_with_custom_interface_for_offset_;
   std::string hardware_system_2dof_standard_interfaces_with_custom_interface_for_offset_missing_;
+  std::string valid_urdf_ros2_control_system_robot_with_gpio_;
 };
 
+TEST_F(TestGenericSystem, valid_urdf_ros2_control_system_robot_with_gpio_)
+{
+  auto urdf = ros2_control_test_assets::urdf_head + valid_urdf_ros2_control_system_robot_with_gpio_ +
+              ros2_control_test_assets::urdf_tail;
+  hardware_interface::ResourceManager rm(urdf);
+
+  // check is hardware is configured
+  std::unordered_map<std::string, rclcpp_lifecycle::State> states_map;
+  states_map = rm.get_components_states();
+  EXPECT_EQ(
+    states_map["GenericSystem2dof"].label(), hardware_interface::lifecycle_state_names::INACTIVE);
+
+  // Check initial values
```

Comment:
Please add here also call the "standard" tests to be sure everything is working also when GPIOs are used.

Reply:
added.